pkg/ops/miningManager.go

```diff
@@ -77,70 +119,272 @@ func CreateMiningManager(ctx context.Context, exitCh chan os.Signal, submitter t
 // Start will start the mining run loop.
 func (mgr *MiningMgr) Start(ctx context.Context) {
 	mgr.Running = true
-	go func(ctx context.Context) {
-		cfg := config.GetConfig()
+	ticker := time.NewTicker(mgr.cfg.MiningInterruptCheckInterval.Duration)
 
-		ticker := time.NewTicker(cfg.MiningInterruptCheckInterval.Duration)
+	if mgr.cfg.RequestData > 0 {
+		if err := mgr.dataRequester.Start(ctx); err != nil {
+			mgr.log.Error("error starting the data requester error:%v", err)
+		}
+	}
 
-		//if you make these buffered, think about the effects on synchronization!
-		input := make(chan *pow.Work)
-		output := make(chan *pow.Result)
-		if cfg.RequestData > 0 {
-			if err := mgr.dataRequester.Start(ctx); err != nil {
-				fmt.Println("error starting the data requester", err)
+	// Start the mining group.
+	go mgr.group.Mine(mgr.toMineInput, mgr.solutionOutput)
+
+	for {
+		select {
+		// Boss wants us to quit for the day.
+		case <-mgr.exitCh:
+			mgr.Running = false
+			return
+		// Found a solution.
+		case solution := <-mgr.solutionOutput:
+			// There is no new challenge so resend any pending solution.
+			if solution == nil {
```

Comment:
when would solution be nil here?

Reply:
`mgr.tasker.GetWork` sends nil to indicate that the challenge hasn't changed this signals the manager that can use the pendingSolution.

Here:
https://github.com/tellor-io/TellorMiner/pull/228/files?w=1#diff-455c0953444badebf81c8379c2da7d4ad3ba73c891e49ab10a4e13bd2a2a68b5R209